packages/block-library/src/image/edit.native.js

```diff
@@ -39,6 +40,44 @@ export default function ImageEdit( { attributes } ) {
 				resizeMethod="scale"
 				source={ { uri: url } }
 			/>
+			{ CaptionTextInput( setAttributes, caption, isSelected ) }
 		</View>
 	);
 }
+
+const CaptionTextInput = ( setAttributes, caption, isSelected ) => {
+	const captionString = [] + caption; // caption can be a string or an array.
+
+	if ( ! captionString.length && ! isSelected ) {
+		return null;
+	}
+
+	const textContainer = ( children ) => {
+		return (
+			<View style={ { padding: 12, flex: 1 } }>
+				{ children() }
+			</View>
+		);
+	};
+
+	const textInput = () => {
+		return (
+			<TextInput
+				style={ { textAlign: 'center' } }
+				underlineColorAndroid="transparent"
+				value={ captionString }
+				placeholder={ 'Write caption…' }
+				onChangeText={ ( newCaption ) => setAttributes( { caption: newCaption } ) }
+			/>
+		);
+	};
+
+	const text = () => {
+		return ( <Text style={ { textAlign: 'center' } }>{ captionString }</Text> );
+	};
+
+	if ( isSelected ) {
+		return textContainer( textInput );
+	}
+	return textContainer( text );
```

Comment:
Any particular reason we fall back to plain textview instead of just letting the editing component in?

Reply:
I tried using the `editable={ isSelected }` property on `TextInput`. It had these issues:

When the block is **not** selected, with a caption written, and I tap over it to select the block, 
- **IOS**: The tap is consumed by the TextInput component and it doesn't select the block.
- **Android**: That property seems to not work and it starts editing the caption anyway, without actually selecting the block.

I tested this approach and it worked as expected, so I decided to keep it. If there's a better way I'll be happy to test it!

Reply:
Yeah, I think the focus/selection issues are already there for the rest of the blocks anyway and we will probably tackle them in unison.

Besides, the textview only solution "solves" the tap-to-focus-block but needs an extra tap to place the cursor.

I think leaving the TextInput (or RichText as the web side does) should be fine for this iteration and should simplify the PR diff for now. WDYT?

Reply:
Sounds good!
And yes! it will get quite a lot simpler. 😁